pricegraph/src/lib.rs

```diff
@@ -9,3 +9,35 @@ mod data;
 
 pub use encoding::*;
 pub use orderbook::Orderbook;
+
+/// A struct representing a transitive orderbook for a base and quote token.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct TransitiveOrderbook {
+    /// Transitive "ask" orders, i.e. transitive orders buying the base token
+    /// and selling the quote token.
+    pub asks: Vec<TransitiveOrder>,
+    /// Transitive "bid" orders, i.e. transitive orders buying the quote token
+    /// and selling the base token.
+    pub bids: Vec<TransitiveOrder>,
+}
+
+/// A struct representing a transitive order for trading between two tokens.
+///
+/// A transitive order is defined as the transitive combination of multiple
+/// orders into a single equivalent order. For example consider the following
+/// two orders:
+/// - *A*: buying 1_000_000 token 1 selling 2_000_000 token 2
+/// - *B*: buying 4_000_000 token 2 selling 1_000_000 token 3
+///
+/// We can define a transitive order *C* buying 1_000_000 token 1 selling
+/// 500_000 token 3 by combining *A* and *B*. Note that the sell amount of token
+/// 3 is limited by the token 2 capacity for this transitive order.
```

Comment:
Awesome description and example to go along with it. Is there any benefit to having so many zeros here? Looks like we could factor out 10^5 here...

Reply:
I kept it there because of things like the minimum traded amount. However, I can just express them in base units (so write it like `1.0`, etc.).